File: apzkr-pzpi-21-11-kokhanevych-pavlo/Task1-Server/src/main/java/com/pet/hotel/data/database/entities/RoomEntity.java
package com.pet.hotel.data.database.entities;

import jakarta.persistence.*;
import lombok.*;
import org.springframework.stereotype.Component;

import java.util.List;

@Data
@Entity
@Table(name = "room")
@NoArgsConstructor
@Component
public class RoomEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "room_id")
    @Setter(AccessLevel.PUBLIC)
    private int roomId;

    @Column(name = "number", nullable = false)
    private int number;

    @Column(name = "area", nullable = false)
    private float area;

    @Column(name = "price_hour", nullable = false)
    private float priceHour;

    @Column(name = "ip", nullable = false)
    private String ip;

    @Column(name = "port", nullable = false)
    private String port;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "hotel_id", nullable = false)
    private HotelEntity hotel;

    @OneToMany(fetch = FetchType.EAGER, mappedBy = "room")
    private List<RentEntity> rents;

    public RoomEntity(int number, float area, float priceHour, String ip, String port, HotelEntity hotel, List<RentEntity> rents) {
        this.number = number;
        this.area = area;
        this.priceHour = priceHour;
        this.ip = ip;
        this.port = port;
        this.hotel = hotel;
        this.rents = rents;
    }
}

File: apzkr-pzpi-21-11-kokhanevych-pavlo/Task4-MobileClient/src/main/java/com/animal/hotel/data/user/entities/RoomEntity.kt
package com.animal.hotel.data.user.entities

import com.animal.hotel.domain.models.user.Room

data class RoomEntity(
     val roomId: Int,
    val number: Int,
    val area: Float,
     val priceHour: Float,
    val ip: String,
    val port: String,
    val hotel: HotelEntity
) {
    fun toRoom(): Room {
        return  Room(roomId, number, area, priceHour, ip, port, hotel.toHotel())
    }

    companion object {
        fun toRoomEntity(room: Room) =  RoomEntity (
            room.roomId,
            room.number,
            room.area,
            room.priceHour,
            room.ip,
            room.port,
            HotelEntity.toHotelEntity(room.hotel)
        )
    }
}
